pkg/multiTenancyPlugins/naming/nameScoping.go

```diff
@@ -129,3 +145,181 @@ func (nameScoping *DefaultNameScopingImpl) Handle(command utils.CommandEnum, clu
+				containerRefIndex := strings.Index(envElement, "affinity:container==") + len("affinity:container==")
+				containerRef := envElement[containerRefIndex:]
+				affinityFromEnvMap[containerRef] = &affinityRefType{AFFINITY_CONTAINER, envElementIndex}
+			} else { // affinity:<label>:<value>
+				labelRefIndex := strings.Index(envElement, "affinity:") + len("affinity:")
+				containerRef := envElement[labelRefIndex:]
+				affinityFromEnvMap[containerRef] = &affinityRefType{AFFINITY_LABEL, envElementIndex}
+
+			}
+		}
+	}
+
+	// create links array
+	linkContainerRefs := make([]string, 0)
+	for _, link := range containerConfig.HostConfig.Links {
+		// link in form [container reference]:[alias]
+		// : and alias are optional
+		containerRef_alias := strings.SplitN(link, ":", 2)
+		// linkSplit[0] == container reference
+		linkContainerRefs = append(linkContainerRefs, strings.TrimSpace(containerRef_alias[0]))
+	}
+
+	var err error
+	var containerReferenceToIdMap map[string]string
+	containerRefs := make([]string, 0)
+	containerRefs = append(containerRefs, containerConfig.HostConfig.VolumesFrom...)
+	containerRefs = append(containerRefs, linkContainerRefs...)
+	if containerReferenceToIdMap, err = getIDsFromContainerReferences(cluster, tenantId, containerRefs, affinityFromEnvMap); err != nil {
+		return err
+	}
+	// ******************update containerConfig******************
+	// update VolumesFrom
+	for i, k := range containerConfig.HostConfig.VolumesFrom {
+		containerConfig.HostConfig.VolumesFrom[i] = containerReferenceToIdMap[k]
+	}
+
+	// update affinity
+	for containerRef, affinityRef := range affinityFromEnvMap {
+		if affinityRef.affinityElementType == AFFINITY_CONTAINER {
+			containerConfig.Env[affinityRef.envElementIndex] = "affinity:container==" + containerReferenceToIdMap[containerRef]
+		} else {
+			containerConfig.Env[affinityRef.envElementIndex] = "affinity:container==" + containerReferenceToIdMap[containerRef]
+		}
+	}
+
+	// update links
+	// We want to create an array of links with no duplicates.  Ideally, to do this we would use a set however
+	// go does not support a native set structure.  Rather we use a map, named linkSet, to accumulated non duplicate links.
+	// Only the keys are important in linkSet;  the values are meaningless.
+	// Once linkSet is created we generate the links array from linkSet.
+	links := make([]string, 0)
+	// We want to generate a set of links with no dugo does not support a native set, s
+	linkSet := make(map[string]string)
+	for _, link := range containerConfig.HostConfig.Links {
+		containerRef_alias := strings.SplitN(link, ":", 2)
+		containerIdName := strings.TrimSpace(containerRef_alias[0])
+		containerId := containerReferenceToIdMap[containerIdName]
+		linkSet[containerId] = ""
+		if containerId != containerIdName {
+			linkSet[containerId+":"+containerIdName] = ""
+		}
+		if len(containerRef_alias) > 1 {
+			linkSet[containerId+":"+strings.TrimSpace(containerRef_alias[1])] = ""
+		}
+	}
```

Comment:
maybe you should define links here and then you can define with the correct length (according to linkSet)